src/core/search/compressed_sorted_set.cc

```diff
@@ -239,7 +240,17 @@ absl::Span<uint8_t> CompressedSortedSet::WriteVarLen(IntType value, absl::Span<u
 std::pair<CompressedSortedSet::IntType, size_t> CompressedSortedSet::ReadVarLen(
     absl::Span<const uint8_t> source) {
   uint64_t out;
-  size_t read = base::flit::ParseT(source.data(), &out);
+  size_t read;
+  // We need this because ParseT reads 8 bytes but source can be less than that
+  // due to the encoding and we end up accessing an invalid memory location
+  if (source.size() < 8) {
+    VarintBuffer ranged_source{0};
+    memcpy(&ranged_source, source.data(), source.size());
+    absl::Span<const uint8_t> source2 = absl::MakeSpan(ranged_source);
+    read = base::flit::ParseT(source2.data(), &out);
```

Comment:
Good catch, it's actually written in the flit comments. You don't need a span to get a pointer, you can pass directly